application-ckeditor-plugins/src/main/resources/xwiki-macro/macroEditor.js

```diff
@@ -683,13 +702,49 @@ define(
         macroEditor.empty().addClass('loading')
           .attr('data-macroId', macroId)
           .prop('requestNumber', requestNumber);
+
+        // Load the macro descriptor
         macroService.getMacroDescriptor(macroId)
           .done(maybeCreateMacroEditor.bind(macroEditor, requestNumber, macroCall))
-          .fail(maybeShowError.bind(macroEditor, requestNumber));
+          .fail(maybeShowError.bind(macroEditor, requestNumber, 'descriptorRequestFailed'));
       }
     };
   },
 
+  load = function(input, macroEditor) {
+    var macroEditorAPI = macroEditor.data('macroEditorAPI');
+    macroEditor.data('hiddenMacroParameters', input.hiddenMacroParameters || []);
+    var macroCall = input.macroCall || {
+      name: input.macroId,
+      parameters: {}
+    };
+    // We need to obey the specified macro identifier in case the user has just changed the macro.
+    macroCall.name = input.macroId || macroCall.name;
+    if (!macroEditorAPI) {
+      // Initialize the macro editor.
+      macroEditor.on('ready', function(event) {
+        macroEditorAPI.focus();
+        submitButton.prop('disabled', false);
+      });
+      macroEditorAPI = macroEditor.xwikiMacroEditor(macroCall, input.syntaxId);
+    } else {
+      macroEditorAPI.update(macroCall, input.syntaxId);
+    }
+  },
+
+  install = function(input, macroEditor) {
+    var macroId = input.macroId + '' + input.syntaxId;
+    var requestNumber = (macroEditor.prop('requestNumber') || 0) + 1;
+
+    macroEditor.empty().addClass('loading')
+      .attr('data-macroId', macroId)
+      .prop('requestNumber', requestNumber);
+
+    macroService.installMacro(input.extensionId, input.extensionVersion)
+      .done(load.bind(macroEditor, input, macroEditor))
```

Comment:
```suggestion
      .done(load.bind(null, input, macroEditor))
```
I don't see any usage of ``this`` inside the ``load`` function so I'd rather keep it ``null``.